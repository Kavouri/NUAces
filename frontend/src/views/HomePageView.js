import React from 'react';
import PropTypes from 'prop-types';

const HomePageView = ({ partners }) => {
  return partners ? partners.map(partner => <li key={partner.id}>{partner.name}</li>) : <div />;
};

HomePageView.propTypes = {
  partners: PropTypes.arrayOf(PropTypes.object),
};

HomePageView.defaultProps = {
  partners: [],
};

export default HomePageView;
