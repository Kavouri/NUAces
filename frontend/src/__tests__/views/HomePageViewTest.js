import React from 'react';
import renderer from 'react-test-renderer';
import HomePageView from '../../views/HomePageView';

describe('HomePageView', () => {
  it('matches a snapshot if partners are not present', () => {
    const tree = renderer.create(<HomePageView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches a snapshot if partners are present', () => {
    const tree = renderer.create(<HomePageView partners={[{ name: 'partner1', id: 1 }, { name: 'partner2', id: 2 }]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
