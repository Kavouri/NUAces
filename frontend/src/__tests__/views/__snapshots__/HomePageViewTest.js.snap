// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageView matches a snapshot if partners are not present 1`] = `null`;

exports[`HomePageView matches a snapshot if partners are present 1`] = `
Array [
  <li>
    partner1
</li>,
  <li>
    partner2
</li>,
]
`;
